src/signature.rs

```diff
@@ -246,7 +241,14 @@ pub fn verify(alg: &VerificationAlgorithm, public_key: untrusted::Input,
               msg: untrusted::Input, signature: untrusted::Input)
               -> Result<(), ()> {
     init::init_once();
-    alg.implementation.verify(public_key, msg, signature)
+    match alg {
```

Comment:
One of the main goals is to make it so, for example, if you only use Ed25519, then the compiler/linker won't bring along ECDSA and RSA with it. This is why we never switch on enums like this, because it makes the dead code elimination in the compiler nearly impossible. So, unless we have clear evidence that the compiler consistently optimizes away the unused cases in this match, I'd definitely like to find a different solution to this.